ycmd/wsgi_server.py

```diff
@@ -15,27 +15,34 @@
 # You should have received a copy of the GNU General Public License
 # along with ycmd.  If not, see <http://www.gnu.org/licenses/>.
 
-from waitress.server import TcpWSGIServer
+from wsgiref.simple_server import WSGIServer, WSGIRequestHandler
+from socketserver import ThreadingMixIn
 import select
 
 
-class StoppableWSGIServer( TcpWSGIServer ):
+class StoppableWSGIServer( ThreadingMixIn, WSGIServer ):
   """StoppableWSGIServer is a subclass of the TcpWSGIServer Waitress server
   with a shutdown method. It is based on StopableWSGIServer class from webtest:
   https://github.com/Pylons/webtest/blob/master/webtest/http.py"""
```

Comment:
Update the comment, or better remove it
```suggestion
  """StoppableWSGIServer extends WSGIServer by adding a shutdown method."""
```